hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/BlockStreamModule.java

```diff
@@ -28,30 +28,28 @@ public interface BlockStreamModule {
 
     @Provides
     @Singleton
-    static BlockNodeConfigExtractor provideBlockNodeConfigExtractor(@NonNull final ConfigProvider configProvider) {
-        final var blockStreamConfig = configProvider.getConfiguration().getConfigData(BlockStreamConfig.class);
-        if (blockStreamConfig.streamToBlockNodes()) {
-            return new BlockNodeConfigExtractorImpl(blockStreamConfig.blockNodeConnectionFileDir());
-        } else {
-            return new NoOpBlockNodeConfigExtractor();
-        }
+    static BlockBufferService provideBlockBufferService(
+            @NonNull final ConfigProvider configProvider, @NonNull final BlockStreamMetrics blockStreamMetrics) {
+        return new BlockBufferService(configProvider, blockStreamMetrics);
     }
 
     @Provides
     @Singleton
-    static BlockStreamStateManager provideBlockStreamStateManager(@NonNull final ConfigProvider configProvider) {
-        return new BlockStreamStateManager(configProvider);
+    static BlockNodeConnectionManager provideBlockNodeConnectionManager(
+            @NonNull final ConfigProvider configProvider,
+            @NonNull final BlockBufferService blockBufferService,
+            @NonNull final BlockStreamMetrics blockStreamMetrics) {
+        final BlockNodeConnectionManager manager = new BlockNodeConnectionManager(
+                configProvider, blockBufferService, blockStreamMetrics, Executors.newSingleThreadScheduledExecutor());
+        blockBufferService.setBlockNodeConnectionManager(manager);
+        return manager;
     }
 
     @Provides
     @Singleton
-    static BlockNodeConnectionManager provideBlockNodeConnectionManager(
-            @NonNull final BlockNodeConfigExtractor blockNodeConfigExtractor,
-            @NonNull final BlockStreamStateManager blockStreamStateManager) {
-        final BlockNodeConnectionManager manager =
-                new BlockNodeConnectionManager(blockNodeConfigExtractor, blockStreamStateManager);
-        blockStreamStateManager.setBlockNodeConnectionManager(manager);
-        return manager;
+    static BlockStreamMetrics provideBlockStreamMetrics(
+            @NonNull final NodeInfo selfNodeInfo, @NonNull final Metrics metrics) {
+        return new BlockStreamMetrics(metrics, selfNodeInfo);
```

Comment:
_Nit_ - here are likely other places Dagger2 will automatically construct a singleton of this type if you annotate it `@Singleton` and annotate the constructor with `@Inject`.